### PR TITLE
Remove span from VLink

### LINKED ISSUE #1840: Audit all the `VLinks` to make sure all have `href` and make it required
## Problem

Noted in https://github.com/WordPress/openverse-frontend/pull/1829#discussion_r975118023 : currently, the VLink component can sometimes not have an `href` prop and be rendered as a span.

## Description

We should only use `VLink` when `href` is available, and remove the `span` fallback from `VLink` component.

## Additional context

More discussion in  https://github.com/WordPress/openverse-frontend/pull/1829

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1840 by @obulat

## Description

This PR removes `span` as the fallback element that was rendered by `VLink` if the `href` prop wasn't provided.
It also adds checks for `href` in all components that use a dynamic `href` prop, and adds a wrapper to the `VContentLink` to make the disabled component render a div instead of an `a` link without an `href`.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/components/VLink.vue [hunk 1]
```diff
@@ -33,19 +31,6 @@
...
-    v-bind="$attrs"
-    :class="{ 'inline-flex flex-row items-center gap-2': showExternalIcon }"
-  >
-    <slot /><VIcon
-      v-if="showExternalIcon && !isInternal"
-      :icon-path="externalLinkIcon"
-      class="inline-block"
-      :size="4"
-      rtl-flip
-    />
```
Comment (dhruvkb) [inline src/components/VLink.vue:47]: Is the removal of `<VIcon/>` accidental or intentional?

**Comment by zackkrida** [comment]
My understanding was that `a` tags without an `href` attribute is a valid and encouraged way to disable a link, particularly paired with `aria-disabled` (and actually, now I'm seeing `role=link` should be used too):

- https://a11y-guidelines.orange.com/en/articles/disable-elements/#disable-a-link
- https://www.scottohara.me/blog/2021/05/28/disabled-links.html

File: src/components/VLink.vue (hunk 1 above)
Comment (obulat) [reply src/components/VLink.vue:47]: Only the Icon inside the span I'd removed, <a> and NuxtLink still have it.

**Comment by obulat** [comment]
> My understanding was that `a` tags without an `href` attribute is a valid and encouraged way to disable a link, particularly paired with `aria-disabled` (and actually, now I'm seeing `role=link` should be used too):
> 
> - https://a11y-guidelines.orange.com/en/articles/disable-elements/#disable-a-link
> - https://www.scottohara.me/blog/2021/05/28/disabled-links.html
> 

Oh, this is interesting, thank you for links, Zack! I tried tabbing to the disabled content links on staging - and couldn't. Those links are skipped when using the Tab key. However, with VO, if you use Ctrl Shift arrow key - you do get to the disabled link! 
With this, I realized that the keyboard access order when using the VoiceOver key with arrows and when using Tab key are different! And the change in tabbing order we implemented ( from the header to the sidebar ) doesn't work with the VO navigation keys 😞

**Comment by zackkrida** [comment]
@obulat do you think we should remove the `div` then and replace with `a`, or a different approach?

File: src/components/VLink.vue [hunk 2]
```diff
@@ -1,27 +1,21 @@
...
     @click.native="$emit('click', $event)"
   >
-    <slot /><VIcon
-      v-if="showExternalIcon && !isInternal"
-      :icon-path="externalLinkIcon"
-      class="inline-block"
-      :size="4"
-      rtl-flip
-    />
+    <slot />
```
Comment (obulat) [inline src/components/VLink.vue:10]: I realized just now that a NuxtLink is only used for the internal links, so it cannot use the `externalLinkIcon`.

**Comment by obulat** [comment]
@sarayourfriend, I would really appreciate your opinion on this approach for disabling links, and not having `span` as the `VLink` fallback.

File: src/components/VLink.vue (hunk 2 above)
Comment (sarayourfriend) [reply src/components/VLink.vue:10]: Interesting. How do we define an "external" link? When we've removed the iframe, will w.org links be "external"? If all links that don't use `NuxtLink` are external then we can remove the conditional `showExternalIcon && !isInternal` from `VIcon`. If it's only things that are not either (a) Openverse frontend links or (b) w.org links, then we could detect it and automatically add the icon without having to rely on the prop.

Probably an additional tightening up here and clarification around the iframe project and the idea of "externality". We typically concern ourselves with this (I think) due to differing privacy expectations on other websites. Given Openverse and WordPress.org follow the same privacy policies and are all controlled by WPF, we probably don't need the external icon for w.org links either. But again, something to clarify and probably something for another issue and to rope the w.org infrastructure folks in on.

**Comment by sarayourfriend** [comment]
If you're asking from an a11y perspective, I don't know enough to comment. The behaviour you're seeing with VO is fine. Tab is only used to move between actionable items (buttons, form elements, links) so it naturally skips over disabled ones. The VO text navigation on the other hand is used to read each element of the page one by one. Tab navigation is probably more likely used by a sighted keyboard user who will have the visual context to see that a link is disabled. Maybe some style elements would need to show this if it's anything beyond a loading state so that all sighted users who are not relying on a screen reader that will relay the aria attribute context to them will understand why something that appears to be a link is not clickable.

These are all problems that already exist if we use a span fallback, so I don't think they need to be solved or debated too much here. I haven't tested the code here so I won't stick an approval on it, but as long as other folks have tested and find it functional, I don't see any a11y reason not to merge this. There is no regression from the previous behaviour and in fact screen readers will now have _more_ context than they did before with the `span` approach.

**Review by dhruvkb — APPROVED** [review]
I'm not a 100% clear on the nuances of a11y for this change, but I tested the following and they work.

- navigation with keyboard
- announcements by VoiceOver

**Review by zackkrida — APPROVED** [review]
LGTM!

**Pull request merged**